docs/setup/telemetry.mdx

```diff
@@ -0,0 +1,113 @@
+
+## What is OpenTelemetry?
+
+[OpenTelemetry](https://opentelemetry.io/) is an open-source observability framework that provides tools, APIs, and SDKs to help instrument, generate, collect, and export telemetry data (metrics, logs, and traces) for analysis. OpenTelemetry is a Cloud Native Computing Foundation (CNCF) project designed to make observability a built-in feature of cloud-native applications.
+
+## Telemetry in Saleor architecture
+
+Saleor's telemetry system is primarily located in the `saleor.core.telemetry` package and consists of the following key components:
+
+### Tracing
+
+Tracing is provided through the `Tracer` class which is a wrapper around OpenTelemetry's tracing capabilities. It allows you to:
+
+- Create spans to measure operation execution time
+- Add attributes and events to spans
+- Establish parent-child relationships between spans
+- Link spans across asynchronous operations
+
+Saleor implements two distinct scopes for traces:
+- **CORE**: For internal system operations and core functionality
+- **SERVICE**: For business logic and service-level operations
+
+What is being traced:
+
+- **HTTP requests** including their duration, full URL, request method, client's IP address, and length of response.
+- **GraphQL queries** including their duration, query string, and execution errors.
+- **GraphQL resolvers** including their duration, field name, parent type, and execution errors.
+- **Database queries** including their duration and the SQL statement.
+
+### Metrics
+
+Metrics are handled through the `Meter` class, which is a wrapper around OpenTelemetry's metrics capabilities. It supports:
+
+- Counters
+- Histograms
+- Up/Down Counters
+
+Like traces, metrics are also separated into CORE and SERVICE scopes.
+
+### Context propagation
+
+Saleor implements context propagation to ensure telemetry context is maintained across:
+- Synchronous operations
+- Asynchronous Celery tasks (via [Span Links](https://opentelemetry.io/docs/concepts/signals/traces/#span-links))
+- Web requests
+
+Context propagation between Saleor and external services is handled through OpenTelemetry's [W3C Trace Context](https://www.w3.org/TR/trace-context/) standard.
+
+## Setting up your own observability
+
+If you're running your own Saleor instance, you can set up telemetry collection and viewing using standard OpenTelemetry tooling. Below is an example of a observability setup where Saleor is configured to send telemetry data to an [OpenTelemetry Collector](https://opentelemetry.io/docs/collector/), which then exports the data to a [Jaeger](https://www.jaegertracing.io/) instance for visualization.
+
+### 1. Configure an OpenTelemetry Collector
+
+Set up an OpenTelemetry Collector to receive, process, and export telemetry data.
```

Comment:
Would be good to mention, that this is a part of docker-compose file, right?

Reply:
When running a collector, this is a standard config file that you need. I can make it more explicit.

Reply:
Added info that this is collector config and a link to docs on configuring collectors.